### ISSUE TITLE
Bug: [no-unnecessary-template-expression] Should not report when expression includes comment

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10434]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=AYKAJA3g9AVABASwHYBcCmqEHskEMA2cAxlgLakYpwxRwDkdAviAHRvADcQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK18iALbF4QifQAexaIMocm6KImjQO0SODABfEEaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
`
${/* intentional comment */ ''}
...`;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-template-expression": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error, as there's no other way to include the comment there.

### Actual Result

```plaintext
Template literal expression is unnecessary and can be simplified. 2:1 - 2:32
```

### Additional Info

Discovered in TypeHero, thanks for letting me do user research on you @bautistaaa 😄: https://github.com/typehero/typehero/blob/0b7a25895c6ffc3707fd1a6348b31f2d38b81b95/tooling/scripts/dl-monaco.ts#L85-L86

```ts
${/* needs to import from monaco-editor to access the Environment type */ ''}
declare let MonacoEnvironment: import("monaco-editor").Environment | undefined;
```

**Comment by kirkwaiblinger** [comment]
Just wondering, would it a viable alternative to 
- downgrade the fix to a suggestion
- make it possible to use a disable comment, like
   ```ts
   `
   ${
     // elsint-disable-next-line @typescript-eslint/no-unnecessary-template-expression
     /* intentional comment */ ''
   }
   ...`;
   ```
   
The reason I ask is that I'm starting to think https://github.com/typescript-eslint/typescript-eslint/issues/8597 was a mistake. And I'm wondering whether reporting here in the first place or whether the inability to disable the report is the bigger problem.  Or whether both should be changed 🤷‍♂️

**Comment by JoshuaKGoldberg** [comment]
> downgrade the fix to a suggestion

I don't want it to report on this _at all_, though 😞 

> make it possible to use a disable comment, like ... #8597

I think that reverting 8597 would be a band-aid fix here. There's no other way to place comments inside template literal strings without getting them into the string. I think this is a valid enough use case that the rule just shouldn't flag in the first place.

**Comment by kirkwaiblinger** [comment]
I'm fine with that 👍 We can disable the report entirely if comments are present to fix this issue. And consider reverting #8597 separately

**Comment by controversial** [comment]
Just ran into this and went to open an issue and found that you beat me to it by a day!

I think it would make sense for this rule to ignore the specific pattern of “comment + empty string” in a template literal expression, either by default or as a configurable option on the rule

**Comment by kirkwaiblinger** [comment]
I've realized there's no need to revert #8597 after we resolve this, since _any_ comment will effectively be a disable comment. Neat 👍
